### ISSUE TITLE
Application crashes at startup with LogBack binding

### ISSUE THREAD

**Issue by Maev4l** [SAP/cf-java-logging-support#14]
Dear colleagues,

When I push my app into CF, it crashes with the following error messages:
2017-09-22T11:07:11.23+0200 [APP/PROC/WEB/0] OUT 09:07:11,218 |-ERROR in ch.qos.logback.core.joran.util.PropertySetter@26336332 - A "com.sap.hcp.cf.logback.encoder.JsonEncoder" object is not assignable to a "ch.qos.logback.core.encoder.Encoder" variable.
2017-09-22T11:07:11.23+0200 [APP/PROC/WEB/0] OUT 09:07:11,218 |-ERROR in ch.qos.logback.core.joran.util.PropertySetter@26336332 - The class "ch.qos.logback.core.encoder.Encoder" was loaded by 
2017-09-22T11:07:11.23+0200 [APP/PROC/WEB/0] OUT 09:07:11,219 |-ERROR in ch.qos.logback.core.joran.util.PropertySetter@26336332 - [ParallelWebappClassLoader
2017-09-22T11:07:11.23+0200 [APP/PROC/WEB/0] OUT   context: ROOT
2017-09-22T11:07:11.24+0200 [APP/PROC/WEB/0] OUT   delegate: false
2017-09-22T11:07:11.24+0200 [APP/PROC/WEB/0] OUT ----------> Parent Classloader:
2017-09-22T11:07:11.24+0200 [APP/PROC/WEB/0] OUT java.net.URLClassLoader [id=17300,parents=System@14021]

My logback.xml file:
```xml
<configuration debug="false" scan="false">
  <appender name="STDOUT-JSON" class="ch.qos.logback.core.ConsoleAppender">
    <encoder class="com.sap.hcp.cf.logback.encoder.JsonEncoder" />
  </appender>
  <!-- for local development, you may want to switch to a more human-readable 
    layout -->
  <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
    <encoder>
      <pattern>%date %-5level [%thread] - [%logger] [%mdc] - %msg%n
      </pattern>
    </encoder>
  </appender>
  <root level="${LOG_ROOT_LEVEL:-WARN}">
    <!-- Use 'STDOUT' instead for human-readable output -->
    <appender-ref ref="STDOUT-JSON" />
  </root>
  <!-- request metrics are reported using INFO level, so make sure the instrumentation 
    loggers are set to that level -->
  <logger name="com.sap.hcp.cf" level="INFO" />
</configuration>
```

My pom file:
```xml
<dependency>
      <groupId>com.sap.hcp.cf.logging</groupId>
      <artifactId>cf-java-logging-support-logback</artifactId>
      <version>2.0.10</version>
    </dependency>
<dependency>
      <groupId>ch.qos.logback</groupId>
      <artifactId>logback-classic</artifactId>
      <version>1.1.11</version>
</dependency>
```
I use logback 1.1.11 as I want to be on par with one pulled by Spring Boot

Any help is appreciated.

Regards,

**Comment by KarstenSchnitter** [comment]
Dear Maevl4,

this looks like a version conflict between logback versions. `cf-java-logging-support-logback` is build against `logback` version 1.1.3. However, this dependency is managed with the _provided_ flag of Maven. Therefore it will pick up the version, that is already used by your application.

We set up the following test scenario: 
  - `cf-java-logging-support-logback` at version 2.0.10
  - `logback` at version 1.1.11

Using this configuration with a very basic app, that does not contain Spring Boot, we were unable to reproduce the behaviour you described. Everything was working fine.

At the moment, we are unsure, what caused the problem on your end. During our investigation we noticed, that you truncated the error messages prematurely. There should have been information on the ClassLoader, that loaded `com.sap.hcp.cf.logback.encoder.JsonEncoder`. We would propose to check your Maven dependency tree for version conflicts.

Regards,
Karsten and Christoph
